src/components/StatePicker.js

```diff
@@ -1,67 +1,106 @@
-import _ from 'underscore';
-import React, {forwardRef} from 'react';
+import React, {useState, useCallback, useMemo} from 'react';
+import {View} from 'react-native';
+import {useRoute, useFocusEffect} from '@react-navigation/native';
 import PropTypes from 'prop-types';
-import Picker from './Picker';
+import useNavigationStorage from '../hooks/useNavigationStorage';
+import compose from '../libs/compose';
+import withNavigation from './withNavigation';
+import sizes from '../styles/variables';
+import styles from '../styles/styles';
 import withLocalize, {withLocalizePropTypes} from './withLocalize';
+import MenuItemWithTopDescription from './MenuItemWithTopDescription';
+import Navigation from '../libs/Navigation/Navigation';
+import ROUTES from '../ROUTES';
+import FormHelpMessage from './FormHelpMessage';
 
 const propTypes = {
-    /** The label for the field */
-    label: PropTypes.string,
-
-    /** A callback method that is called when the value changes and it receives the selected value as an argument. */
-    onInputChange: PropTypes.func.isRequired,
-
-    /** The value that needs to be selected */
-    value: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
-
-    /** The ID used to uniquely identify the input in a Form */
-    inputID: PropTypes.string,
-
-    /** Saves a draft of the input value when used in a form */
-    shouldSaveDraft: PropTypes.bool,
-
-    /** Callback that is called when the text input is blurred */
-    onBlur: PropTypes.func,
+    /** Current State from user address  */
+    stateISO: PropTypes.string,
 
     /** Error text to display */
     errorText: PropTypes.string,
 
+    /** Default value to display */
+    defaultValue: PropTypes.string,
+
     ...withLocalizePropTypes,
 };
 
 const defaultProps = {
-    label: '',
-    value: undefined,
+    stateISO: '',
     errorText: '',
-    shouldSaveDraft: false,
-    inputID: undefined,
-    onBlur: () => {},
+    defaultValue: '',
 };
 
-const StatePicker = forwardRef((props, ref) => {
-    const STATES = _.chain(props.translate('allStates'))
-        .sortBy((state) => state.stateName.toLowerCase())
-        .map((state) => ({value: state.stateISO, label: state.stateName}))
-        .value();
+function BaseStatePicker(props) {
+    const route = useRoute();
```

Comment:
I need it because the bank flow uses a `policyID` in the URL and the `Navigation.getActiveRoute()` removes any query param, so I need the whole path so I can go back, if not the wrong screen will be displayed

Reply:
Resolved